### DISCUSSION TITLE
Whats the best way to clear all selections

### DISCUSSION THREAD

**Discussion by KumarilDave** [xyflow/xyflow#3172] [Q&A] [answered]
Hello folks. I am migrating away from react diagram to react flow. React diagram had to util to clear all the selection. Is there a similar method in react flow that can help clear up all the the selections (nodes and edges). If not can you recommend the most performant way to do it ?

**Comment by bcakmakoglu** [comment, marked as answer]
1. `useStore` provides a `unselectNodesAndEdges` function

Given you have wrapped your App in the `ReactFlowProvider` you can use the `useStore` hook, with which you can access the store directly and use `unselectNodesAndEdges`

```ts
// our selector for the `useStore` hook, receives an obj of type `ReactFlowState` as the fn arg
const selector = (state: ReactFlowState) => {
  return {
    unselectAll: state.unselectNodesAndEdges
  };
};


const Flow = () => {
  const [nodes, setNodes, onNodesChange] = useNodesState(initialNodes);
  const [edges, setEdges, onEdgesChange] = useEdgesState(initialEdges);

  const { unselectAll } = useStore(selector);

  // ...

```

[Here's an example.](https://codesandbox.io/s/vibrant-johnson-fxypw4?file=/src/Flow.tsx:925-1142)

2. `setNodes` and `setEdges`

You can simply call these setter actions to unselect all nodes and edges.

```ts
setNodes((nodes) => nodes.map((n) => ({ ...n, selected: false })));

setEdges((edges) => edges.map((e) => ({ ...e, selected: false })));
```
